internal/exec/stages/files/passwd.go

```diff
@@ -66,12 +66,23 @@ func (s *stage) createPasswd(config types.Config) error {
 		}
 
 		s.relabel(deglobbed...)
-		s.relabel(
-			"/etc/.pwd.lock",
-			// for OSTree-based systems
-			"/var/home",
-			"/var/roothome",
-		)
+		s.relabel("/etc/.pwd.lock")
+		for _, user := range config.Passwd.Users {
+			if user.NoCreateHome != nil && *user.NoCreateHome == true {
+				continue
+			}
+			if user.Name == "root" {
+				rootdir_deglobbed, err := s.expandGlobList("/root/*")
+				if err != nil {
+					return err
+				}
+				s.relabel(rootdir_deglobbed...)
+			} else if user.HomeDir != nil && *user.HomeDir != "" {
```

Comment:
AFAIK it's legal to change root's home directory with an Ignition config, so we should do this test first.